src/net/sourceforge/kolmafia/textui/RuntimeLibrary.java

```diff
@@ -8121,6 +8128,19 @@ public static Value get_locket_monsters(ScriptRuntime controller) {
     return value;
   }
 
+  public static Value get_permed_skills(ScriptRuntime controller) {
+    AggregateType type = new AggregateType(DataTypes.BOOLEAN_TYPE, DataTypes.SKILL_TYPE);
+    MapValue value = new MapValue(type);
+
+    for (var permedSkill : KoLConstants.permedSkills) {
+      var skill = DataTypes.makeSkillValue(permedSkill.getSkillId(), true);
+      var hardcore =
+          DataTypes.makeBooleanValue(KoLConstants.hardcorePermedSkills.contains(permedSkill));
```

Comment:
As the only thing you're doing with this list is checking whether an element is within it, it may as well be a `Set<int>`.